xapian-core/weight/tfidfweight.cc

```diff
@@ -136,7 +142,8 @@ TfIdfWeight::get_maxpart() const
 {
     Xapian::termcount wdf_max = get_wdf_upper_bound();
     Xapian::termcount len_min = get_doclength_lower_bound();
-    double wdfn = get_wdfn(wdf_max, len_min, normalizations[0]);
+    Xapian::termcount len_max = get_doclength_upper_bound();
+    double wdfn = get_wdfn(wdf_max, len_min, len_max, normalizations[0]);
     return get_wtn(wdfn * idfn, normalizations[2]) * wqf_factor;
```

Comment:
I think you just want to pass `len_min` for both.  For the 'P' case we want to minimise `doclen`, so we have to pass `len_min` for `doclen`.  Only 'L' currently looks at `uniqterms` and for that we just need to pass the same value for `doclen` and `uniqterms` - for 'L'  `get_wdfn()` only divides one by the other, so it only matters that they are equal (and non-zero).

(More generally, I doubt there will always be single a set of parameters we can just call `get_wdfn()` with to get an upper bound for any plausible wdf norm, but for the set of wdf norms implemented so far it seems there is).